### PR TITLE
Add Enum support for request/response messages

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Fix the issue: https://github.com/ballerina-platform/ballerina-lang/issues/10038

## Automation tests
 - Integration tests
   Contains an integration test.

## Samples
grpc service
```
endpoint grpc:Listener listener {
    host:"localhost",
    port:8555
};

map<orderInfo> ordersMap;

type orderInfo record {
    string id;
    Mode mode;
};

public type Mode "r";
@final public Mode READ = "r";

@grpc:ServiceConfig
service testEnumService bind listener {
    testEnum(endpoint caller, orderInfo orderReq) {
        string permission;
        match orderReq.mode {
            READ => permission = "r";
        }
        _ = caller->send(permission);
        _ = caller->complete();
    }
}
```
Generated proto file.
```
syntax = "proto3";
package grpcservices;
import "google/protobuf/wrappers.proto";
service testEnumService {
		rpc testEnum(orderInfo) returns (google.protobuf.StringValue);
}
message orderInfo {
	string id = 1;
	Mode mode = 2;
}
enum Mode {
	r = 0;
}
```
grpc client
```
public function main (string... args) {

       endpoint testEnumServiceBlockingClient blockingEp {
        url:"http://localhost:8555"
    };

    orderInfo orderReq = {id:"100500", mode:r};
    var addResponse = blockingEp->testEnum(orderReq);
    match addResponse {
        (string, grpc:Headers) payload => {
            string result;
            (result, _) = payload;
            log:printInfo("Response - " + result + "\n");        }
        error err => {
            log:printError("Error from Connector: " + err.message + "\n");
        }
    }
}
```

### CODE REVIEW

File: tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    public function getCallerActions () returns testEnumServiceStub {
+        return self.stub;
+    }
+};
+
+
+type orderInfo record {
+    string id;
+    Mode mode;
+
```
Comment (ldclakmal) [inline tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal]: Shall we remove the new line here as well?

File: tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    public grpc:Stub stub;
+
+    function initStub (grpc:Client ep) {
+        grpc:Stub navStub = new;
+        navStub.initStub(ep, "blocking", DESCRIPTOR_KEY, descriptorMap);
+        self.stub = navStub;
+    }
+
+    function testEnum (orderInfo req, grpc:Headers? headers = ()) returns ((string, grpc:Headers)|error) {
+
```
Comment (ldclakmal) [inline tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal]: Shall we remove the new line?

File: stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/Field.java [hunk 3]
```diff
@@ -68,6 +68,10 @@ public String getDefaultValue() {
         public Field build() {
             String fieldType = FIELD_TYPE_MAP.get(fieldDescriptor.getType()) != null ? FIELD_TYPE_MAP.get
                     (fieldDescriptor.getType()) : fieldDescriptor.getTypeName();
+            if (fieldType.startsWith(".")) {
+                String[] fieldTypeArray = fieldType.split(("\\."));
```
Comment (Manuri) [inline stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/builder/components/Field.java]: Have we not used these constants in other places? Might be better to introduce constants.

File: tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/grpc;
+
+function testEnum() returns (string) {
+    endpoint testEnumServiceBlockingClient blockingEp {
+        url:"http://localhost:8555"
+    };
+
+    orderInfo orderReq = {id:"100500", mode:r};
```
Comment (Manuri) [inline tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal]: Formatting issue?

File: tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    public grpc:Stub stub;
+
+    function initStub (grpc:Client ep) {
+        grpc:Stub navStub = new;
+        navStub.initStub(ep, "non-blocking", DESCRIPTOR_KEY, descriptorMap);
+        self.stub = navStub;
+    }
+
+    function testEnum (orderInfo req, typedesc listener, grpc:Headers? headers = ()) returns (error?) {
+
```
[the remark this answers is not in the document]
Comment (Bhashinee) [reply tests/ballerina-integration-test/src/test/resources/grpc/clients/grpc_enum_test_client.bal]: This is the generated stub for the client.

**Pull request merged**